pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -203,7 +208,30 @@ func init() {
 }
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
-func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {}
+func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {
+	if vs.cfg == nil {
+		return
+	}
+
+	// Only on controller node it is required to register listeners.
+	var err error
+	//clientBuilder := clientBuilder
+	client, err := clientBuilder.Client("vSphere-cloud-provider")
+	if err != nil {
+		glog.Errorf("failed to create kubeclient: %v", err)
+      return
+	}
+
+	// Register callbacks for node updates
+	factory := informers.NewSharedInformerFactory(client, 5*time.Minute)
+	nodeInformer := factory.Core().V1().Nodes()
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    vs.NodeAdded,
+		DeleteFunc: vs.NodeDeleted,
+	})
+	go nodeInformer.Informer().Run(wait.NeverStop)
+   glog.V(4).Infof("vSphere cloud provider initialized")
```

Comment:
The formatting seems to be off. Can you for this file?

Reply:
Done